src/Raven.NewClient/Document/Async/AsyncDocumentSession.cs

```diff
@@ -26,7 +29,8 @@ namespace Raven.NewClient.Client.Document.Async
     /// </summary>
     public partial class AsyncDocumentSession : InMemoryDocumentSessionOperations, IAsyncDocumentSessionImpl, IAsyncAdvancedSessionOperations, IDocumentQueryGenerator
     {
-        private readonly AsyncDocumentKeyGeneration asyncDocumentKeyGeneration;
+        private readonly AsyncDocumentKeyGeneration _asyncDocumentKeyGeneration;
+        private readonly Lazy<OperationExecuter> _operations;
```

Comment:
Why is this lazy? We have to create it per session, even if it isn't required. And we don't need to thread safety guarantees since a session is single threaded.